biz.aQute.bndlib/src/aQute/bnd/build/Workspace.java

```diff
@@ -1541,18 +1560,38 @@ public String _findproviders(String[] args) throws Exception {
 
 	static final String _findprovidersHelp = "${findproviders;<namespace>;<filter>}";
 
+	/**
+	 * A macro that returns a set of resources in bundle selection format from
+	 * the repositories, but excludes the workspace. For example:
+	 *
+	 * <pre>
+	 * ${findproviders;osgi.wiring.package;(osgi.wiring.package=aQute.bnd.build)}
+	 * </pre>
+	 */
+
+	public String _findexternalproviders(String[] args) throws Exception {
```

Comment:
Not sure `external` is the right name here. Do we even need a new macro? Couldn't we just add another optional argument to the existing macro?

`${findproviders;<namespace>;<filter>[;<excludeWorkspace>]}`

Where `excludeWorkspace` defaults to false and behaves like the current impl. When set to `true` it does not search the workspace.

I think this is much better and avoids a new doc page too.

Reply:
> Where `excludeWorkspace` defaults to false and behaves like the current impl. When set to `true` it does not search the workspace.

Actually, instead of a boolean argument, we should do an "enum" like STRATEGY in the repo macro: <https://bnd.bndtools.org/macros/repo.html>.  Valid values for the enum could be: `REPOS` and `WORKSPACE` where `WORKSPACE` is the default.

Reply:
Good Idea. I ended up with 3 Strategies: ALL, WORKSPACE and REPOS. WORKSPACE will only look in the Workspace. Might be useful as well.